src/installer/managed/Microsoft.NET.HostModel/MachO/CodeDirectoryHeader.cs

```diff
@@ -0,0 +1,121 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.InteropServices;
+
+namespace Microsoft.NET.HostModel.MachO;
+
+/// <summary>
+/// Version 0x20400
+/// </summary>
+[StructLayout(LayoutKind.Sequential)]
+internal struct CodeDirectoryHeader
+{
+    private BlobMagic _magic;
+    private uint _size;
+    private CodeDirectoryVersion _version;
+    private CodeDirectoryFlags _flags;
+    private uint _hashesOffset;
+    private uint _identifierOffset;
+    private uint _specialSlotCount;
+    private uint _codeSlotCount;
+    private uint _executableLength;
+    public byte HashSize;
+    public HashType HashType;
+    public byte Platform;
+    public byte Log2PageSize;
+    private uint _reserved;
+    private uint _scatterOffset;
+    private uint _teamIdOffset;
+    private uint _reserved2;
+    private ulong _codeLimit64;
+    private ulong _execSegmentBase;
+    private ulong _execSegmentLimit;
+    private ExecutableSegmentFlags _execSegmentFlags;
+
+    public BlobMagic Magic
+    {
+        get => (BlobMagic)((uint)_magic).ConvertFromBigEndian();
```

Comment:
I found it easier to keep the memory representation identical to what's in the file so that reading/writing the file is simple. I don't see much of a benefit of changing it personally.